src/reader/tf_reader.py

```diff
@@ -73,24 +81,26 @@ def read(self, epoch_number):
                 dataset = dataset.shuffle(buffer_size=self.shuffle_size)
         if self.prefetch:
             dataset = dataset.prefetch(buffer_size=self.prefetch_size)
-        self._dataset = dataset.batch(self.batch_size)
+        self._dataset = dataset.batch(self.batch_size, drop_remainder=True)
+
 
     def next(self):
         """
         Provides the iterator over tfrecord data pipeline.
         :return: data to be processed by the training step.
         """
         super().next()
-        a = iter(self._dataset)
-        count = 1
-        total = math.ceil(self.num_samples*self.num_files/self.batch_size/self.comm_size)
-        for i in a:
-            progress(count, total, "Reading TFRecord Data")
-            count += 1
-            yield i
-            yield next(a)
-            if count > total:
-                break
+        dataset = self._dataset
+
+        # In tf, we can't get the length of the dataset easily so we calculate it
+        if self._debug:
+            total = math.ceil(self.num_samples*self._local_file_list_size/self.batch_size/self.comm_size)
+            logging.debug(f"{utcnow()} Rank {self.my_rank} should read {total} batches")
+
+        # The previous version crashed when all workers could not generate the same amount of batches
+        # Using the inbuilt tensorflow dataset iteration seems to work fine, was there an advantage of doing it the old way?
```

Comment:
To have consistent iteration logic for each case. If we do TensorFlow iteration for one vs our own for others its would be inconsistent for format comparison.

Reply:
Maybe add a issue for this as enhancement.